fairseq2n/src/fairseq2n/data/concat_data_source.cc

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "fairseq2n/data/concat_data_source.h"
+#include <vector>
+
+namespace fairseq2n::detail {
+
+concat_data_source::concat_data_source(
+    std::vector<data_pipeline> &&pipelines)
+    : pipelines_{std::move(pipelines)}
+{}
+
+std::optional<data>
+concat_data_source::next()
+{
+    std::optional<data> d;
+    for (auto &p : pipelines_) {
+        d = p.next();
+        if (d) {
```

Comment:
Nit: for single line statement blocks, we do not use opening/closing braces. You can write this as:

```cpp
if (d)
  return d;
```

Reply:
Fixed